libp2p/upgrademngrs/muxedupgrade.nim

```diff
@@ -86,22 +86,21 @@ method upgrade*(
   return muxer
 
 proc new*(
-  T: type MuxedUpgrade,
-  muxers: seq[MuxerProvider],
-  secureManagers: openArray[Secure] = [],
-  ms: MultistreamSelect): T =
-
+    T: type MuxedUpgrade,
+    muxers: seq[MuxerProvider],
+    secureManagers: openArray[Secure] = [],
+    ms: MultistreamSelect): T =
   let upgrader = T(
     muxers: muxers,
     secureManagers: @secureManagers,
     ms: ms)
 
-  upgrader.streamHandler = proc(conn: Connection) {.async.} =
+  upgrader.streamHandler = proc(conn: Connection) {.async: (raises: []).} =
     trace "Starting stream handler", conn
     try:
       await upgrader.ms.handle(conn) # handle incoming connection
     except CancelledError as exc:
-      raise exc
+      return
```

Comment:
why has this been changed?

Reply:
`streamHandler` is launched using `asyncSpawn`, and raising an error from `asyncSpawn` crashes the entire app (defect).